src/services/logger/Logger.ts

```diff
@@ -0,0 +1,28 @@
+export const logEvents: Record<string, string> = {
+  PAGINATION_PREVIOUS_BUTTON_CLICKED: 'pagination-previous-button-clicked',
+  PAGINATION_NEXT_BUTTON_CLICKED: 'pagination-next-button-clicked',
+  SIDEBAR_ITEM_CLICKED: 'sidebar-item-clicked',
+  TABLE_REFRESH_BUTTON_CLICKED: 'table-refresh-button-clicked',
+  TABS_TAB_CHANGE: 'tabs-tab-change',
+  CREATE_DATA_PLANE_PROXY_CLICKED: 'create-data-plane-proxy-clicked',
+}
+
+type LogArgs = [string] | [string, object | undefined] | [string, object | undefined, Error | undefined]
+
+export default class Logger {
+  isEnabled: boolean = true
```

Comment:
I don't understand why this is here when you have a specialised class `DisabledLogger` to me `DisabledLogger` would override the `_log()` method with an empty body and you wouldn't need this here right? If not what's the point of `DisabledLogger` ?

Reply:
That’s also an option, but if we add more methods to `Logger`, you’d always need to override them all in `DisabledLogger`.

Reply:
The way these log classes are typically implemented is by having 1 internal methods and all the others just call this 1 method.
To me either you delete the `DisabledLogger` class because it's not useful or you make it as mentioned previously. This seems to be a little confusing...

Reply:
That works for me. Overriding `_log` in the disabled logger should be enough as well.

Reply:
This was changed in https://github.com/kumahq/kuma-gui/pull/1003/commits/ee9829d1de23ca8b2a8c7c1722398565796a9cc1.